### ISSUE TITLE
Errors when using bdist_wheel and pip install with --use-pep517 or pyproject.toml

### ISSUE THREAD

**Issue by f-cozzocrea** [hpyproject/hpy#435]
I'm seeing an issue when whenever I'm trying to install an extension with the `setup.py bdist_wheel + pip install` method with a very basic pyproject.toml file that looks like this:
```
[project]
name = "quickstart"
version = "0.0.1"
```

This is the error output:
```
error: Error: setup script specifies an absolute path:

    /home/f-cozzocrea/projects/hpy-pep517/venv/hpy-pep517/lib/python3.10/site-packages/hpy/devel/src/runtime/argparse.c

setup() arguments must *always* be /-separated paths relative to the
setup.py directory, *never* absolute paths.
```

I've created a basic repository here with steps to reproduce the errors: https://github.com/f-cozzocrea/hpy-pep517

--- System info ---
Hpy version: 0.9.0rc2
CPython version: 3.10
OS: Ubuntu 22.04

**Comment by f-cozzocrea** [comment]
It seems like the error is coming from setuptools because absolute paths in a setup script are not compliant with [PEP 517](https://peps.python.org/pep-0517/).

**Comment by fangerer** [comment]
Hi @f-cozzocrea! Thank you again for this issue.
@hodgestar: This is the problem I've mentioned in our last call. Newer setuptools requires to have relative paths in the sources list. Making relative paths to HPy runtime sources doesn't make a lot of sense to me. What do you think? Should we copy the sources to the extension's build dir?

**Comment by hodgestar** [comment]
I think point 1 above is overstated. `hpy.devel` is a build dependency. All of it is by definition outside the package being built, and I doubt "the Python community" (whoever they happen to be) intended to ban build dependencies.

We don't want to compile these for anyone -- they're intended to just be extra C utilities provided for convenience. Extensions should compile them themselves if they use them.

I would vote that we copy the extra sources into the build folder during the build, but also add an option to not copy them in. For this matches our intention most closely -- "here are some useful C utilities that you can include in your extension".

Perhaps the copying could use  `importlib.resources` to do the copying.

@fangerer @f-cozzocrea Does my reasoning make sense to you? If there is some standard way to handle this, it would be good to know, but keeping what little simplicity we can in the `hpy.devel` setup utilities is also quite important.

**Comment by f-cozzocrea** [comment]
@hodgestar I agree with your reasoning. When I say "Python community", I just mean the outcomes of packaging PEPs (like 517) which disallow absolute paths. I think copying by default in a simple and robust way makes the most sense.

**Comment by hodgestar** [comment]
@f-cozzocrea Woot. I agree that the banning by default was probably worth it to avoid people accidentally depending on external files, even if it makes things a bit more complex for us.

**Comment by fangerer** [comment]
Sounds good to me as well.

Adding a bit more background on #310: This issue exists because we have seen compilation problems with some package (I don't remember which one). Since we are adding the HPy helper sources, it might happen that the source code is not compatible with certain compiler options. The idea was to build a static lib with our configuration.

**Comment by paugier** [comment]
I guess we could follow what Pythran (https://github.com/serge-sans-paille/pythran) does. The Pythran package contains a lot of C++ files (in https://github.com/serge-sans-paille/pythran/tree/master/pythran/pythonic), which are used during compilation of C++ extensions produced by Pythran. Pythran is used in Scipy and scikit-image and it supports setuptools and Meson.

**Comment by hodgestar** [comment]
@paugier Do you know what Pythran does? I.e. does it copy the files into the builder folder somehow? Or tell setuptools or meson to do that?

**Comment by paugier** [comment]
I had a look and it seems that Pythran does not include its own sources, which are just used as headers and then users need to use `pythran.get_include` in their `setup.py` or `meson.build` file. Pythran can do that since `pythran.pythonic`, its C++ backend, is a header-only library, with real headers and "headers" containing the implementation.

I don't know if it can be applicable for HPy.

If it not the case, a question to setuptools (https://github.com/pypa/setuptools) could help.
